src/NuGet.Services.Metadata.Catalog.Monitoring/Validation/ValidationFactory.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Net.Http;
+using Autofac;
+using Microsoft.Extensions.Logging;
+using NuGet.Services.Storage;
+using StorageFactory = NuGet.Services.Metadata.Catalog.Persistence.StorageFactory;
+
+namespace NuGet.Services.Metadata.Catalog.Monitoring
+{
+    /// <summary>
+    /// Helper class for constructing validation business logic implementations.
+    /// </summary>
+    public static class ValidationFactory
+    {
+        public static PackageValidator CreatePackageValidator(
+            string galleryUrl,
+            string indexUrl,
+            StorageFactory auditingStorageFactory,
+            ValidatorConfiguration validatorConfig,
+            EndpointConfiguration endpointConfig,
+            Func<HttpMessageHandler> messageHandlerFactory,
+            ILoggerFactory loggerFactory)
+        {
+            if (auditingStorageFactory == null)
+            {
+                throw new ArgumentNullException(nameof(auditingStorageFactory));
+            }
+
+            var builder = new ContainerBuilder();
+
+            builder.RegisterLogging(loggerFactory);
+            builder.RegisterValidatorConfiguration(validatorConfig);
+            builder.RegisterEndpointConfiguration(endpointConfig);
+            builder.RegisterMessageHandlerFactory(messageHandlerFactory);
+            builder.RegisterEndpoints();
+            builder.RegisterSourceRepositories(galleryUrl, indexUrl);
+            builder.RegisterValidators();
+
+            builder
+                .RegisterInstance(auditingStorageFactory)
+                .AsSelf()
+                .As<StorageFactory>();
+
+            builder.RegisterType<PackageValidator>();
+
+            var container = builder.Build();
```

Comment:
The container should be disposed after the work of the job is done.

Reply:
I couldn't see anything in `JsonConfigurationJob` with disposing. Can you explain how I should dispose this container?

Reply:
Spoke offline. Won't fix.

Reply:
After an offline discussion, we discovered a couple things:
1 - We have this problem with several other of our uses of DI containers--for example, `JsonConfigurationJob` does not dispose its containers.
2 - It is nontrivial to refactor the code so that the container can be disposed because of the separation of `Init` and `Run` in our jobs.
3 - The primary purpose of disposing of the container is to dispose of `IDisposable`s in the container.
4 - Fortunately, this job does not put any `IDisposable`s into the container.

I have come to the conclusion that it is not worth fixing this. If any problems come up, we can fix the code then.